store/postgres/replay_repository.go

```diff
@@ -265,3 +265,29 @@ func (repo *replayRepository) GetByProjectIDAndStatus(projectID uuid.UUID, statu
 	}
 	return replaySpecs, nil
 }
+
+func (repo *replayRepository) GetByProjectID(projectID uuid.UUID) ([]models.ReplaySpec, error) {
+	var replays []Replay
+	if err := repo.DB.Preload("Job").Joins("JOIN job ON replay.job_id = job.id").
```

Comment:
Do we need to do this explicetly? I think gorm handles this automatically?

Reply:
yes, we need to add the join statement as preload is not actually joining them, thus cannot use `job.project_id` in the where clause ([reference](https://stackoverflow.com/questions/55656002/how-to-select-by-fields-in-preloaded-object))